File: app/src/main/java/ru/ls/donkitchen/app/AppComponent.java
package ru.ls.donkitchen.app;

import com.squareup.otto.Bus;

import javax.inject.Singleton;

import dagger.Component;
import ru.ls.donkitchen.activity.base.ActivityModule;
import ru.ls.donkitchen.activity.base.ActivitySubComponent;
import ru.ls.donkitchen.rest.Api;

/**
 * A component whose lifetime is the life of the application.
 */
@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {
	void inject(DonKitchenApplication application);

	Bus provideBus();
	Api provideApi();

	ActivitySubComponent plus(ActivityModule module);
//	ServiceSubComponent plus(ServiceModule module);
}

File: app/src/main/kotlin/ru/ls/donkitchen/app/AppComponent.kt
package ru.ls.donkitchen.app

import dagger.Component
import ru.ls.donkitchen.activity.base.ActivityModule
import ru.ls.donkitchen.activity.base.ActivitySubComponent
import ru.ls.donkitchen.service.base.ServiceModule
import ru.ls.donkitchen.service.base.ServiceSubComponent
import ru.ls.donkitchen.ui.receiptdetail.ReceiptDetailActivityModule
import ru.ls.donkitchen.ui.receiptdetail.ReceiptDetailActivitySubComponent
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class)) interface AppComponent {
    fun inject(application: DonKitchenApplication)

    operator fun plus(module: ActivityModule): ActivitySubComponent
    operator fun plus(module: ServiceModule): ServiceSubComponent
//    operator fun plus(module: SplashModule): SplashSubComponent
    operator fun plus(module: ReceiptDetailActivityModule): ReceiptDetailActivitySubComponent
}
